Personalized album quiz and display of top 5 tracks and artists

yBoranbayev opened the pull request:
The demonstration of the implemented feature: https://youtu.be/jxTvfuwMtDM
- Implementation does not use database and at the end of the quiz it will delete the quiz. 
- Quizzes can be access by other user at the moment. 
- Uses previous user interface didn't see any issues with having more than 5 questions per quiz. 
- For implementation used production database, however disabled saving users into database.
- Implemented the display of top 5 tracks and artists 
- [x] #39 
- [x] #40
- [x] #52

yBoranbayev committed:
Changed methods for album cover quiz
added templates for album cover quiz
Need to delete quiz.html static file
Need to modify databases, currently couldn't sign in locally

yBoranbayev committed:
implemented personalized album quizzes
also implemented display of top 5 artists and top 5 tracks
didn't really worked on css
will link a demonstration through youtube to github
used production database
worked locally, didn't check on render

yBoranbayev committed:
minor unnecessary text deletion

yBoranbayev committed:
handled case when there are multiple quizzes generated

sbaljeet13 commented on src/main/java/com/group6/harmoniq/controllers/SpotifyController.java:
```diff
@@ -129,12 +133,35 @@ public String callback(@RequestParam("code") String code, @RequestParam("state")
                     User user = getUserProfile(accessToken);
                     user.setTopArtist(getTopArtist(accessToken));
                     user.setTopTrack(getTopTrack(accessToken));
+                    System.out.println("SpotifyID" + user.getSpotifyId());
+                    System.out.println("DisplayName" + user.getDisplayName());
+                    System.out.println("Email" + user.getEmail());
+                    System.out.println("Followers" + user.getFollowers());
+                    System.out.println("ImageURL" + user.getImageUrl());
+                    System.out.println("TopArtist" + user.getTopArtist());
+                    System.out.println("TopTrack" + user.getTopTrack());
+                    System.out.println("externalSpotifyUrl" + user.getExternalSpotifyUrl());
```
I think this can be removed since it doesn't serve any purpose in the app other than  debugging locally

sbaljeet13 commented on src/main/java/com/group6/harmoniq/controllers/SpotifyController.java:
```diff
@@ -295,4 +322,188 @@ private Track getTopTrack(String accessToken) throws Exception {
         }
     }
+
+    private List<Artist> getTopArtists(String accessToken) throws Exception {
```
The existing `getTopArtist` method can be modified to get the top 10 artists instead of creating a new method

sbaljeet13 commented on src/main/java/com/group6/harmoniq/controllers/SpotifyController.java:
```diff
@@ -295,4 +322,188 @@ private Track getTopTrack(String accessToken) throws Exception {
+        }
+    }
+
+    private List<Track> getTopTracks(String accessToken) throws Exception {
```
Similarly, we can refactor the `getTopTrack` to get the top 10 tracks and avoid the duplicate code

sbaljeet13 commented on src/main/resources/templates/profile.html:
```diff
@@ -34,19 +34,21 @@ <h1 th:text="${user.displayName}"></h1>
+                <div th:each="artist, iterStat : ${topArtists}" class="card" th:if="${iterStat.index < 5}"> 
+                    <a th:href="${artist.spotifyUrl}" target="_blank">
+                        <img th:src="${artist.imageUrl}" th:alt="${artist.name}">
+                        <p class="subtext">TOP ARTIST</p>
```
This would move above the section as a heading and be replaced by the order of the artist in the top artists list

sbaljeet13 commented on src/main/resources/templates/profile.html:
```diff
@@ -34,19 +34,21 @@ <h1 th:text="${user.displayName}"></h1>
+                <div th:each="track, iterStat : ${topTracks}"class="card" th:if="${iterStat.index < 5}">
+                    <a th:href="${track.spotifyUrl}" target="_blank">
+                        <img th:src="${track.albumCoverUrl}" th:alt="${track.name}">
+                        <p class="subtext">TOP TRACKs</p>
```
Similarly, this would move up as a heading of this section and replaced by the order of the track in the top tracks list

sbaljeet13 reviewed (commented):
The album quiz is personalized now! Beautiful! 

It would be better to remove the `System.out.println()` lines and the inline comments because the code the self-explanatory with the use of menanigul variables and method names. We should only use in-line comments rarely to explain some complex logic

yBoranbayev committed:
deleted unnecessary System.out.printlns
deleted the getTopTrack and getTopArtist methods, instead it takes first from the list
changed the profile.html so artists and tracks has heading with indication that shows what placement it has

yBoranbayev commented:
![image](https://github.com/user-attachments/assets/d8668655-29f7-407a-a4e6-a4d30390ad9b)
this is how profile.html page looks like now
Just to be sure, I am using dark mode, so some objects has dark theme, the same thing with video screencast

sbaljeet13 commented:
Something like this: 

<img width="701" alt="image" src="https://github.com/user-attachments/assets/a733fd94-f1e3-4dcc-a02d-7bb09ee5f79b">

With "Top Artists" and "Top Tracks" headings and just the order number on each card

https://www.figma.com/design/6FHvVEdFbCUonR7hDIx4Af/HarmonIQ?node-id=121-36&t=wernSLd6XPf0SEG1-1

yBoranbayev committed:
changed profile page's css (index.css)
added separate css for QuizTypesView.html

yBoranbayev commented:
https://youtu.be/GVTeOqxyxpI
Updated css screen cast

sbaljeet13 committed:
Add Top Artists and Top Tracks Headings

sbaljeet13 committed:
Merge branch 'main' into personalized-album-quiz

yBoranbayev committed:
modified tests according to new Album Quiz

yBoranbayev committed:
added tests for album quiz test

yBoranbayev committed:
Merged main into personalized-album-quiz

yBoranbayev committed:
returned save user to database feature
